src/PageContent/RequestPickup/RecurringPickupRequest.js

```diff
@@ -0,0 +1,284 @@
+import React, {Component} from 'react';
+import { RequestRepeatType, RequestDurationType } from '../../Enums.js';
+import './RequestPickup.css';
+import PickupSummary from './PickupSummary.js';
+
+class RecurringPickupRequest extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            memberList: [
+                {id: 'dhA03LwTp3cibXVUcb3nQqO34wj1', name: 'Test DA1 Member1'}, 
+                {id: 'fbCm3Yrbi4e12WgpVz3gq25VKea2', name: 'Test DA1 Member2'}
+            ],
+            delivererGroups: [
+                {id: 'R8BAHrxdkfQoAmfWvGa1OJmjQP43', name: 'Test DG1'},
+                {id: 'sS4dqgxgLIXtPf42DydgkWLWeHT2', name: 'Test DG2'}
+            ],
+            receivingAgencies: [
+                {id: 'uCm0OG4WeoSyjk3c0rdY3mlaBXl2', name: 'Test RA2'},
+                {id: 'uGOFJ8NqHjbZhKAYzSZFRs1dSKD3', name: 'Test RA1'}
+            ],
+            fields: {},
+            errors: {},
+            isOpen: false,
+            formInfo: [],
+            dayOfWeek: ''
+        }; 
+        this.submitRequest = this.submitRequest.bind(this);
+        this.createRequest = this.createRequest.bind(this);
+        this.toggleModal = this.toggleModal.bind(this);
+    }
+
+    componentDidMount(){
+        // Query DB and setState for memberList, delivererGroups, and receivingAgencies
+    }
+
+    handleValidation(){
+        let fields = this.state.fields;
+        let errors = {};
+        let formIsValid = true;
+
+        //Date
+        if(!fields['startDate']){
+            formIsValid = false;
+            errors['startDate'] = 'Start date cannot be empty';
+        }
+
+        if((fields['recurTimes'] === '' || !fields['recurTimes']) && fields['endDate'] !== '' && fields['endDate'] < fields['startDate']){
```

Comment:
should set `'durationType'` in fields and check against that since it's a radio button, it can only be one or the other. The user can technically still fill in `recurTimes` even if they chose "End by".

you can group all `'endDate'` / `'recurTimes'` checks into:

```
if (!fields['durationType']) {
    // the they need to have one or the other error msg
} else {
    if (fields['durationType'] === RequestDurationType.DATE) {
        // perform all endDate related checks
    } else {
        // perform all recurTimes related checks
    }
}
```